### PR TITLE
Use structured logging

### PR DESCRIPTION
**Pull request by mmzeeman**
This PR changes log messages to structured logging format.

- [x] Transform things to structured logging calls
- [x] Transform yecc errors to maps to improve readability. (See: https://www.erlang.org/doc/man/yecc.html)

### CODE REVIEW

File: src/template_compiler.erl [hunk 1]
```diff
@@ -196,7 +196,7 @@ block_lookup({ok, TplFile}, BlockMap, ExtendsStack, DebugTrace, Options, Vars, R
             case lists:member(Module, ExtendsStack) of
                 true ->
                     FileTrace = [Module:filename() | [ M:filename() || M <- ExtendsStack ]],
-                    ?LOG_ERROR("[template_compiler] Template recursion: ~p", [FileTrace]),
+                    ?LOG_ERROR(#{ text => "Template recursion.", trace => FileTrace}),
```
Comment (mworrell) [inline src/template_compiler.erl:199]: Just an idea, could we transform the trace to a stack trace format?  Then the logger can show it as a stack trace.

File: src/template_compiler.erl [hunk 2]
```diff
@@ -355,16 +355,14 @@ compile_forms(Filename, Forms) ->
...
                     {ok, Module};
                 Error ->
-                    ?LOG_ERROR("Error loading compiling forms for ~p: ~p",
-                                [Filename, Error]),
+                    ?LOG_ERROR(#{ text => "Error loading compiling forms.", filename => Filename, error => Error}),
                     Error
             end;
         error ->
-            ?LOG_ERROR("Error compiling forms for ~p", [Filename]),
+            ?LOG_ERROR(#{ text => "Error compiling forms.", filename => Filename}),
```
Comment (mworrell) [inline src/template_compiler.erl:362]: In the other logging I always added `result => error`  and then, if known, `reason => ...`

File: src/template_compiler.erl (hunk 1 above)
Comment (mmzeeman) [reply src/template_compiler.erl:199]: Probably.. I want to change the yecc errors top maps as well. That will probably improve the readability. Not sure what a filetrae looks like tbh.

Comment (mworrell) [reply src/template_compiler.erl:199]: Would be great, the Yecc errors are quite confusing. We should be able to see in one go what the problem is and where.

File: src/template_compiler_admin.erl [hunk 3]
```diff
@@ -87,11 +87,12 @@ compile_file(Filename, TplKey, Options, Context) ->
                             % io:format("Error compiling template ~p: ~p:~n~p at~n ~p~n",
                             %           [Filename, What, Error, Stack]),
                             ?LOG_ERROR(
-                                "Error compiling template ~p: ~p: ~p",
-                                [Filename, What, Error],
                                 #{
-                                    template => Filename,
-                                    stack => Stack
+                                  text =>"Error compiling template.",
```
Comment (vkatsuba) [inline src/template_compiler_admin.erl:91]:
```suggestion
                                  text => "Error compiling template.", 
```

File: src/template_compiler.erl [hunk 4]
```diff
@@ -355,16 +355,14 @@ compile_forms(Filename, Forms) ->
                 {module, _Module} ->
                     {ok, Module};
                 Error ->
-                    ?LOG_ERROR("Error loading compiling forms for ~p: ~p",
-                                [Filename, Error]),
+                    ?LOG_ERROR(#{ text => "Error loading compiling forms.", filename => Filename, error => Error}),
```
Comment (vkatsuba) [inline src/template_compiler.erl:358]:
```suggestion
                    ?LOG_ERROR(#{ text => "Error loading compiling forms.", filename => Filename, error => Error }),
```

File: src/template_compiler.erl (hunk 2 above)
Comment (vkatsuba) [inline src/template_compiler.erl:362]:
```suggestion
            ?LOG_ERROR(#{ text => "Error compiling forms.", filename => Filename }),
```

File: src/template_compiler.erl [hunk 5]
```diff
@@ -355,16 +355,14 @@ compile_forms(Filename, Forms) ->
...
                     Error
             end;
         error ->
-            ?LOG_ERROR("Error compiling forms for ~p", [Filename]),
+            ?LOG_ERROR(#{ text => "Error compiling forms.", filename => Filename}),
             {error, {compile, []}};
         {error, Es, Ws} ->
-            ?LOG_ERROR("Errors compiling ~p: ~p  (warnings ~p)",
-                        [Filename, Es, Ws]),
+            ?LOG_ERROR(#{ text => "Errors compiling.", filename => Filename, errors => Es, warnings => Ws}),
```
Comment (vkatsuba) [inline src/template_compiler.erl:365]:
```suggestion
            ?LOG_ERROR(#{ text => "Errors compiling.", filename => Filename, errors => Es, warnings => Ws }),
```

File: src/template_compiler_admin.erl [hunk 6]
```diff
@@ -220,8 +221,7 @@ handle_cast(Msg, State) ->
 handle_info({'DOWN', MRef, process, _Pid, Reason}, State) ->
     case lists:keytake(MRef, 1, State#state.compiling) of
         {value, {_MRef, TplKey, _From}, Compiling1} ->
-            ?LOG_ERROR("[template_compiler] Process compiling ~p down (reason ~p), restarting other waiter",
-                        [TplKey, Reason]),
+            ?LOG_ERROR(#{ text => "Process compiling down. Restarting other waiter.", tpl_key => TplKey, reason => Reason}),
```
Comment (vkatsuba) [inline src/template_compiler_admin.erl:224]:
```suggestion
            ?LOG_ERROR(#{ text => "Process compiling down. Restarting other waiter.", tpl_key => TplKey, reason => Reason }),
```

**Comment by mworrell** [comment]
Rebased on master

**Pull request merged**